src/algorithm/minkowski/gjk/simplex/simplex3d.rs

```diff
@@ -96,6 +121,50 @@ where
 }
 
 #[inline]
+fn get_closest_point_on_face<S>(
+    a: &Vector3<S>,
+    b: &Vector3<S>,
+    c: &Vector3<S>,
+    normal: &Vector3<S>,
+) -> Vector3<S>
+where
+    S: BaseFloat,
+{
+    use {Continuous, Plane, Ray3};
+    let ap = Point3::from_vec(*a);
+    let bp = Point3::from_vec(*b);
+    let cp = Point3::from_vec(*c);
+    let ray = Ray3::new(Point3::origin(), -*normal);
+    // unwrapping is safe, because the degenerate face will have been removed by the outer algorithm
+    let plane = Plane::from_points(ap, bp, cp).unwrap();
+    match plane.intersection(&ray) {
+        Some(point) => {
+            let (u, v, w) = barycentric_vector(point.to_vec(), *a, *b, *c);
+            if in_range(u) && in_range(v) && in_range(w) {
+                // inside face, use face
+                point.to_vec()
+            } else {
+                // NOTE: this should never be hit, because the outer algorithm should have
```

Comment:
perhaps, we should turn this `in_range` checks into an assert with this message then?